HT5.py

```diff
@@ -0,0 +1,111 @@
+import datetime
+import random
+
+
+class Publication:
+    def __init__(self, name):
+        self.name = name
+
+    def get_publication_text(self):
+        txt = input('Type text: ')
+        return txt
+
+    def write_to_list(self):
+        pub_list.insert(0, pub.name)
+        pub_list.insert(1, self.get_publication_text())
+        return pub_list
+
+    def write_to_file(self, l1):
+        file = open('myfile.txt', 'a+')
+        file.write('---------------------------------' + '\n')
+        file.write(l1[0] + '\n' + l1[1] + '\n' + l1[2] + '\n')
+        file.write('---------------------------------' + '\n\n')
+
+
+class News(Publication):
+    def __init__(self, name, publish_datetime):
+        Publication.__init__(self, name=name)
+        self.publish_datetime = publish_datetime
+
+    def get_publish_date(self):
+        return self.publish_datetime.strftime('%Y-%m-%d %H:%M')
+
+    def news_location(self):
+        location = input('Type city name: ')
+        return location
+
+    def write_to_list_news(self):
+        pub_list.append(inp_location + ', ' + self.get_publish_date())
+        return pub_list
+
+
+class Advertising(Publication):
+    def __init__(self, name):
+        Publication.__init__(self, name=name)
+
+    def due_date(self):
+        print('Type advertising expiration date')
+        exp_year = int(input('Enter a year: '))

```

Comment:
What if user input text? or random number like 999999? Users can be unique persons:) I think we need to handle it:)

Reply:
Python returns an error: ValueError: year 999999 is out of range. Do we need additional checks here?

Reply:
Errors are causing script to fail, I'm suggesting to add the logic for handing these cases without script fails

Reply:
ok